### PR TITLE
python3Packages.django-ninja: init at 0.22.2

### PR DESCRIPTION
**Pull request by elohmeier**
## Description of changes

Added django-ninja package.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/django-ninja/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    flit
+    psycopg2
+    pytest-asyncio
+    pytest-django
+    pytestCheckHook
+  ];
+
+  meta = with lib; {
+    changelog = "https://github.com/vitalik/django-ninja/releases/tag/v${version}";
+    description = "Web framework for building APIs with Django and Python type hints.";
```
Comment (natsukium) [inline pkgs/development/python-modules/django-ninja/default.nix:37]:
```suggestion
    description = "Web framework for building APIs with Django and Python type hints";
```
Please remove the trailing period.

File: pkgs/development/python-modules/django-ninja/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+buildPythonPackage rec {
+  pname = "django-ninja";
+  version = "0.22.2";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "vitalik";
+    repo = "django-ninja";
+    rev = "v${version}";
+    sha256 = "sha256-oeisurp9seSn3X/5jFF9DMm9nU6uDYIU1b6/J3o2be0=";
```
[the remark this answers is not in the document]
Comment (elohmeier) [reply pkgs/development/python-modules/django-ninja/default.nix]: Thanks, updated!

File: pkgs/development/python-modules/django-ninja/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  };
+
+  propagatedBuildInputs = [ django pydantic ];
+
+  nativeCheckInputs = [
+    flit
```
[the remark this answers is not in the document]
Comment (elohmeier) [reply pkgs/development/python-modules/django-ninja/default.nix]: Done.

File: pkgs/development/python-modules/django-ninja/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+, pydantic
+, pytest-asyncio
+, pytest-django
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "django-ninja";
+  version = "0.22.2";
+  format = "pyproject";
```
[the remark this answers is not in the document]
Comment (elohmeier) [reply pkgs/development/python-modules/django-ninja/default.nix:17]: Added.

**Review by natsukium — APPROVED** [review]
thanks

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 255088` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.django-ninja</li>
    <li>python310Packages.django-ninja.dist</li>
    <li>python311Packages.django-ninja</li>
    <li>python311Packages.django-ninja.dist</li>
  </ul>
</details>

**Pull request merged**